### PR TITLE
fix(vscode): Use single source of truth for project file consistency checks (.vscode, local.settings.json, host.json)

### PR DESCRIPTION
**Pull request by andrew-eldridge**
## Commit Type

- [ ] feature - New functionality
- [x] fix - Bug fix
- [ ] refactor - Code restructuring without behavior change
- [ ] perf - Performance improvement
- [ ] docs - Documentation update
- [ ] test - Test-related changes
- [ ] chore - Maintenance/tooling

## Risk Level

- [ ] Low - Minor changes, limited scope
- [x] Medium - Moderate changes, some user impact
- [ ] High - Major changes, significant user/system impact

## What & Why
Introduces new single source of truth for .vscode files (tasks.json, launch.json, settings.json, extensions.json). Previously there were multiple implementations of .vscode file generation and re-generation which mostly had the same behavior with some inconsistencies between implementations. Consolidating to simplify the code and prevent these from diverging again.

## Impact of Change

- **Users**: Fixes some edge case bugs in .vscode file generation depending on project type
- **Developers**: Adds central source of truth for each .vscode file
- **System**: N/A

## Test Plan

- [x] Unit tests added/updated
- [ ] E2E tests added/updated
- [x] Manual testing completed
- [ ] Tested in: 

## Contributors
@andrew-eldridge

### CODE REVIEW

File: apps/vs-code-designer/src/app/utils/vsCodeConfig/tasks.ts [hunk 1]
```diff
@@ -140,151 +141,43 @@ async function overwriteTasksJson(context: IActionContext, projectPath: string):
...
+    const files = await fse.readdir(projectPath);
+    const projFileName = files.find((f) => (f.endsWith('.csproj') || f.endsWith('.fsproj')) && f.toLowerCase() !== 'extensions.csproj');
+    if (projFileName) {
+      return await getTargetFramework(new ProjectFile(projFileName, projectPath));
+    }
+  } catch {
+    // Fall through
+  }
+  return undefined;
+}
```
[the remark this answers is not in the document]
Comment (andrew-eldridge) [reply apps/vs-code-designer/src/app/utils/vsCodeConfig/tasks.ts]: should not be a concern for any valid logic app or custom code project

File: apps/vs-code-designer/src/app/commands/initProjectForVSCode/initProjectStepBase.ts [hunk 2]
```diff
@@ -190,7 +190,7 @@ export abstract class InitProjectStepBase extends AzureWizardExecuteStep<IProjec
   }
 
   private insertNewTaskInputs(context: IProjectWizardContext, existingInputs: ITaskInputs[] = [], newInputs: ITaskInputs[]): ITaskInputs[] {
-    if (context.workflowProjectType === WorkflowProjectType.Bundle) {
+    if (context.projectPackageType === ProjectPackageType.Bundle) {
```
[the remark this answers is not in the document]
Comment (andrew-eldridge) [reply apps/vs-code-designer/src/app/commands/initProjectForVSCode/initProjectStepBase.ts]: removed method

File: apps/vs-code-designer/src/app/commands/initProjectForVSCode/initDotnetProjectStep.ts [hunk 3]
```diff
@@ -103,59 +95,16 @@ export class InitDotnetProjectStep extends InitProjectStepBase {
...
-      },
-    ];
+    const { tasks } = generateTasksJson({
+      projectType: ProjectType.logicApp,
+      projectPackageType: ProjectPackageType.Nuget,
+      hasFuncBinaries: binariesExistSync(funcDependencyName),
+      targetFramework: this.targetFramework,
+    });
+    return tasks as TaskDefinition[];
   }
```
[the remark this answers is not in the document]
Comment (andrew-eldridge) [reply apps/vs-code-designer/src/app/commands/initProjectForVSCode/initDotnetProjectStep.ts]: no longer calling generateTasksJson here

File: apps/vs-code-designer/src/app/utils/project.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ */
+export async function detectProjectType(projectPath: string): Promise<ProjectType> {
+  if ((await hasCodefulWorkflowSetting(projectPath)) || (await hasCodefulSdkReference(projectPath))) {
+    return ProjectType.codeful;
+  }
+
+  const customCodeProjects = await tryGetLogicAppCustomCodeFunctionsProjects(projectPath);
+  if (customCodeProjects && customCodeProjects.length > 0) {
+    return ProjectType.customCode;
+  }
```
Comment (lambrianmsft) [inline apps/vs-code-designer/src/app/utils/project.ts:22]: With this, if we have a workspace with a logic app, a logic app with custom code, wouldn't this also return the type for the logic app as a custom code project? Do we now group rules engine logic apps as custom code as well?

Comment (andrew-eldridge) [reply apps/vs-code-designer/src/app/utils/project.ts:22]: if I understand the scenario it's la1: regular logic app, la2: logic app with custom code, func1: custom code functions project belonging to la2. In this case, la2 would give customCode project type and la1 would be logicApp- the tryGetLogicAppCustomCodeFunctionsProjects will return [] for la1 and [func1] for la2

**Review by lambrianmsft — APPROVED** [review]

File: apps/vs-code-designer/src/app/commands/createNewCodeProject/CodeProjectBase/CreateFunctionAppFiles.ts
Comment (andrew-eldridge) [inline apps/vs-code-designer/src/app/commands/createNewCodeProject/CodeProjectBase/CreateFunctionAppFiles.ts:83]: .vscode/* files for custom code functions projects weren't moved to the centralized project consistency checks/file generators (only logic app projects are handled), so issues with invalid .vscode config in these projects won't be caught. May want to address in a future pr

File: apps/vs-code-designer/src/app/utils/codeless/validateProjectArtifacts.ts [hunk 5]
```diff
@@ -398,30 +346,30 @@ async function ensureDesignTimeDirectory(projectPath: string): Promise<Uri> {
  * such as a pinned extension bundle version are not lost.
  * @param {IActionContext} context - The action context.
  * @param {string} projectPath - The logic app project root.
- * @returns {Promise<{ uri: Uri; hostRegenerated: boolean; settingsRegenerated: boolean; changedArtifacts: string[] }>}
+ * @returns {Promise<{ uri: Uri; hostRegenerated: boolean; localSettingsRegenerated: boolean; changedArtifacts: string[] }>}
  * The design-time directory Uri, which baseline files were regenerated, and the human-readable label(s)
  * for the artifact(s) that changed.
  */
 export async function regenerateDesignTimeDirectory(
```
Comment (andrew-eldridge) [inline apps/vs-code-designer/src/app/utils/codeless/validateProjectArtifacts.ts]: this file serves similar purpose to vscodeConsistency for the local.settings.json/host.json files. Should be moved to projectConsistency

**Review by bjbennet — APPROVED** [review]
